src/main/groovy/4configureEc2Plugin.groovy

```diff
@@ -0,0 +1,175 @@
+/**
+*
+* Configure the EC2 plugin
+*
+**/
+import java.util.logging.Logger
+import jenkins.model.Jenkins
+import hudson.model.Node
+import hudson.plugins.ec2.*
+import com.amazonaws.services.ec2.model.InstanceType
+
+@Grapes([
+    @Grab(group='org.yaml', module='snakeyaml', version='1.17')
+])
+import org.yaml.snakeyaml.Yaml
+
+Logger logger = Logger.getLogger("")
+Jenkins jenkins = Jenkins.getInstance()
+Yaml yaml = new Yaml()
+
+String configPath = System.getenv("JENKINS_CONFIG_PATH")
+try {
+    configText = new File("${configPath}/ec2_config.yml").text
+} catch (FileNotFoundException e) {
+    logger.severe("Cannot find config file path @ ${configPath}/ec2_config.yml")
+    jenkins.doSafeExit(null)
+    System.exit(1)
+}
+ec2Config = yaml.load(configText)
+ec2CloudConfig = ec2Config.CLOUDS
+
+// Get the ec2 plugin version
+pluginVersion = Float.parseFloat(jenkins.pluginManager.getPlugin('ec2').getVersion())
+
+List<AmazonEC2Cloud> clouds = new ArrayList<AmazonEC2Cloud>();
+for (cloudConfig in ec2CloudConfig) {
+    List<SlaveTemplate> templates = new ArrayList<SlaveTemplate>();
+    ec2AmiConfig = cloudConfig.AMIS
+    // Loop through the AMIs
+    for (amiConfig in ec2AmiConfig) {
+        // Create Spot Config if applicable
+        maxBid = amiConfig.SPOT_CONFIG.SPOT_MAX_BID_PRICE
+        // The constructor changes based on the plugin so add support for both
+        if (pluginVersion < 1.32) {
```

Comment:
Unfortunately, I suspect we will have to do this type of thing more often, as we upgrade plugins